File: Java/src/World/body/animal/Antelope.java
package World.body.animal;

import GUI.Color_obj;
import World.Point;
import World.World;
import World.body.Animal;
import World.body.Body;

import java.util.concurrent.ThreadLocalRandom;

public class Antelope extends Animal {

    public Antelope(Point point_location, World world) {
        super(point_location, world);
        setName("Antelope");
        setColor(Color_obj.ANTELOPE);
        setInitiative(4);
        setPower(4);
    }
    @Override
    protected void action(){
        Point point = new Point(getPoint_location().getX(), getPoint_location().getY());
        Point new_point = new Point(getPoint_location().getX(), getPoint_location().getY());
        int range = 2;
        while(range > 0){
            random_location(new_point);
            setPoint_location(new_point);
            if(point.getX() == new_point.getX() && point.getY() == new_point.getY()){
                range=2;
            }
            else range--;
        }
        setLast_position(point);
        setPoint_location(new_point);
    }

    @Override
    protected void collision(Body attacker) {
        int randomNumber = ThreadLocalRandom.current().nextInt(1, 101) % 2;
        if(randomNumber==0 || getWorld().free_spaces(this)==0){
            collision(attacker);
        }
        else{
            Point new_point = new Point(getPoint_location().getX(), getPoint_location().getY());
            random_location_empty(this, new_point);
            getWorld().getNews_panel().add_news("Move to ("+getPoint_location().getX()+", "+getPoint_location().getY()+")");
            getWorld().getBoxes()[getPoint_location().getY()-1][getPoint_location().getX()-1].setColor(attacker.getColor());
            setLast_position(getPoint_location());
            setPoint_location(new_point);
            getWorld().getNews_panel().add_news("Run away to ("+getPoint_location().getX()+", "+getPoint_location().getY()+")");
            getWorld().getBoxes()[getPoint_location().getY()-1][getPoint_location().getX()-1].setColor(getColor());
        }
    }

    @Override
    protected boolean repel_attack(Body attacker) {
        return false;
    }
}

File: Python/World/Body/Animal/antelope.py
import random

from GUI.colors import Colors
from World.Body.animal import Animal
from World.point import Point


class Antelope(Animal):
    def __init__(self, point_location, world):
        super().__init__(point_location, world)
        self.set_name("Antelope")
        self.set_color(Colors.ANTELOPE)
        self.set_initiative(4)
        self.set_power(4)

    def action(self):
        point = Point(self.get_point_location().x, self.get_point_location().y)
        new_point = Point(self.get_point_location().x, self.get_point_location().y)
        range_val = 2
        while range_val > 0:
            self.random_location(new_point)
            self.set_point_location(new_point)
            if point.x == new_point.x and point.y == new_point.y:
                range_val = 2
            else:
                range_val -= 1
        self.set_last_position(point)
        self.set_point_location(new_point)

    def collision(self, attacker):
        randomNumber = random.randint(1, 100) % 2
        if randomNumber == 0 or self.get_world().free_spaces(self) == 0:
            self.collision(attacker)
        else:
            new_point = Point(self.get_point_location().x, self.get_point_location().y)
            self.random_location_empty(self, new_point)
            self.world.mainWindow.add_text("Move to (" + str(self.get_point_location().x) +
                                                       ", " + str(self.get_point_location().y) + ")")
            self.get_world().get_boxes[self.get_point_location().y - 1][self.get_point_location().x - 1].set_color(
                attacker.get_color())
            self.set_last_position(self.get_point_location())
            self.set_point_location(new_point)
            self.world.mainWindow.add_text("Run away to (" + str(self.get_point_location().x) +
                                                       ", " + str(self.get_point_location().y) + ")")
            self.get_world().get_boxes[self.get_point_location().y - 1][self.get_point_location().x - 1].set_color(
                self.get_color())

    def repel_attack(self, attacker):
        return False
